spec/system/user_page/user_preferences_security_spec.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+describe "User preferences for Security", type: :system do
+  fab!(:password) { "kungfukenny" }
+  fab!(:email) { "email@user.com" }
+  fab!(:user) { Fabricate(:user, email: email, password: password) }
+  let(:user_preferences_security_page) { PageObjects::Pages::UserPreferencesSecurity.new }
+  let(:user_menu) { PageObjects::Components::UserMenu.new }
+
+  before do
+    user.activate
+    sign_in(user)
+  end
+
+  describe "Security keys" do
+    it "adds a 2F security key and logs in with it" do
+      # simulate browser credential authorization
```

Comment:
This is so cool @pmusaraj I didn't think selenium would support this :exploding_head: